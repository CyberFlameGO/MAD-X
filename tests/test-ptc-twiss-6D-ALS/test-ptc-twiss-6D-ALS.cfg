1-7     *     skip         # head
*       *     any abs=1e-14 rel=1e-16
215     *     abs=5e-15    # beam energy in PTC ???
219     *     abs=3e-13    # machine length
1045    *     skip         # tail
907-947 *     skip         # precision in detecting reference frame differences changes from 32 to 64
