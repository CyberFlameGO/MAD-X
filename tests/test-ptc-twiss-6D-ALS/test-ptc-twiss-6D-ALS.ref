
  ++++++++++++++++++++++++++++++++++++++++++++
  +     MAD-X 5.05.01  (64 bit, Linux)       +
  + Support: mad@cern.ch, http://cern.ch/mad +
  + Release   date: 2019.06.07               +
  + Execution date: 2019.06.27 10:19:08      +
  ++++++++++++++++++++++++++++++++++++++++++++
!example of 6D calculation on ALS (synchrotron light source)

option, -echo;



beam, particle = electron, energy = sqrt(1.5*1.5+emass*emass);  use, period=ALS;



select,flag=ptc_twiss,clear;

select, flag=ptc_twiss, column=name,s,Energy,x,y,pt,t,

                          beta11,beta22,beta33,mu1,mu2,mu3,disp1,disp2,

                          re11,re12,re13,re14,re15,re16,

                          re21,re22,re23,re24,re25,re26,

                          re31,re32,re33,re34,re35,re36,

                          re41,re42,re43,re44,re45,re46,

                          re51,re52,re53,re54,re55,re56,

                          re61,re62,re63,re64,re65,re66;







ptc_create_universe;

Determined SECTOR NMUL MAX :  2
  ptc_create_layout, model=1, method=2, nst=20, exact=false, closed_layout=true;

 MAD-X Beam Parameters
                  Energy : 0.150000E+01
          Kinetic Energy : 0.149949E+01
      Particle Rest Mass : 0.510999E-03
                Momentum : 0.150000E+01
 Setting MADx with 
     energy           1.50000008703997     
     method                   2
     Num. of steps           20
     charge          -1.00000000000000     
  Length of machine:    196.878135791546     
 The machine is a RING
 ------------------------------------ PTC Survey ------------------------------------
 Before start:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
 Before   end:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
 After  start:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
 After    end:   5.691967490256688E-014  0.000000000000000E+000 -7.716050021144838E-015


!  ptc_printframes, file="ALS.C", format=rootmacro;!this prints my visualisation with ROOT



  !!!!!       T I M E   T R U E ,   T O T P A T H  F A L S E

  ptc_setswitch, debuglevel=1, exact_mis=true, time=true, totalpath=false;

 Setting debug level to           1
maxaccel is not present (keeping current value)
exact_mis is found and its value is 1.000000
radiation is not present (keeping current value)
modulation is not present (keeping current value)
stochastic is not present (keeping current value)
envelope is not present (keeping current value)
fringe is not present (keeping current value)
totalpath is found and its value is 0.000000
time is found and its value is 1.000000
nocavity is not present (keeping current value)
seed is not present (keeping current value)
obs_points pro_ptc_setswitch Done
  

!  ptc_script, file="printflatfile.txt";

  

  ptc_twiss, icase=6, no=3, closed_orbit=true, rmatrix, x=0, y=0, t=0.15;

 ************ State Summary ****************
 MADTHICK=>KIND =   32  DRIFT-KICK-DRIFT       
  Rectangular Bend: input arc length (rho alpha) 
  Default integration method     2
  Default integration steps     20
 This is an electron (positron actually if charge=1) 
       EXACT_MODEL =  FALSE
       TOTALPATH   =     0
       RADIATION   =  FALSE
       STOCHASTIC  =  FALSE
       ENVELOPE    =  FALSE
       NOCAVITY    =  FALSE
       TIME        =  TRUE 
       FRINGE      =  FALSE
       PARA_IN     =  FALSE
       ONLY_2D     =  FALSE
       ONLY_4D     =  FALSE
       DELTA       =  FALSE
       SPIN        =  FALSE
       MODULATION  =  FALSE
       RAMPING     =  FALSE
       ACCELERATE  =  FALSE
 Closed orbit:   1.849313249509225E-015 -9.806347110831090E-020  0.000000000000000E+000  0.000000000000000E+000 -0.149896228999759       1.540436433373398E-012

++++++ table: ptc_twiss_summary

            length            alpha_c          alpha_c_p         alpha_c_p2 
       196.8781358     0.001597061569           -1000000           -1000000 

        alpha_c_p3              eta_c           gamma_tr                 q1 
          -1000000     0.001596945515        25.02298816       0.2902019435 

                q2                dq1                dq2                 qs 
      0.2991177483                  0                  0     0.003033858939 

        beta_x_min         beta_x_max         beta_y_min         beta_y_max 
      0.8243387316        12.21021568        1.344962596        22.46553125 

         beta11min          beta11max          beta12min          beta12max 
      0.8243387316        12.21021568    3.209661831e-71    1.948821143e-40 

         beta13min          beta13max          beta21min          beta21max 
    5.33500918e-09     0.003445478433     7.67582947e-36      1.4289069e-34 

         beta22min          beta22max          beta23min          beta23max 
       1.344962596        22.46553125    2.644095966e-35     2.43001136e-31 

         beta31min          beta31max          beta32min          beta32max 
   1.297134175e-07      0.01267018466    1.015602187e-32    1.015971364e-32 

         beta33min          beta33max           disp1min           disp1max 
       16.49298388        16.49448224                  0       0.2383825039 

          disp2min           disp2max           disp3min           disp3max 
     -0.1918836552       0.1918826607   -3.499773087e-16    3.555563493e-16 

          disp4min           disp4max             deltap            orbit_x 
  -1.887918286e-16    1.957412697e-16                  0     1.84931325e-15 

          orbit_px            orbit_y           orbit_py           orbit_pt 
  -9.806347111e-20                  0                  0    1.540436433e-12 

           orbit_t             xcorms             ycorms            pxcorms 
      -0.149896229    1.796834153e-13                  0    1.699545599e-13 

           pycorms             tcorms            ptcorms             xcomax 
                 0        0.149896229    1.540436312e-12    3.671968951e-13 

            ycomax            pxcomax            pycomax             tcomax 
                 0    2.955696519e-13                  0       -0.149896229 

           ptcomax             xcomin             ycomin            pxcomin 
   1.540436433e-12    4.521846686e-16                  0   -2.955710265e-13 

           pycomin             tcomin            ptcomin 
                 0       -0.149896229    1.540403378e-12 
!  ptc_twiss, icase=6, no=2, betx=10, bety=10,betz=10;

  

!  plot, table=ptc_twiss, vaxis=beta11,beta22, haxis=s;

!  plot, table=ptc_twiss, vaxis=mu3, haxis=s;

!  plot, table=ptc_twiss, vaxis=energy, haxis=s;

  

  write, table=ptc_twiss, file="ALS.6D.ptc.twiss";

  write, table=ptc_twiss_summary, file="ALS.6D.ptc.summ.twiss";

  

ptc_end;





!mux_ptc = table(ptc_twiss, ALS$end, mu1);

!muy_ptc = table(ptc_twiss, ALS$end, mu2);



!value, mux_ptc, muy_ptc;

!value, mux_tw, muy_tw;



!plot, table=ptc_twiss, haxis=s, vaxis=beta11,beta22, colour=100;

!plot, table=ptc_twiss, haxis=s, vaxis=disp1, colour=100;

!plot, table=ptc_twiss, haxis=s, vaxis=disp2, colour=100;

!plot, table=ptc_twiss, haxis=s, vaxis=disp3, colour=100;

!plot, table=ptc_twiss, haxis=s, vaxis=disp4, colour=100;

!plot, table=ptc_twiss, haxis=s, vaxis=x, colour=100;



!removefile, file='ALS.C';


  Number of warnings: 0

  ++++++++++++++++++++++++++++++++++++++++++++
  +          MAD-X finished normally         +
  ++++++++++++++++++++++++++++++++++++++++++++
