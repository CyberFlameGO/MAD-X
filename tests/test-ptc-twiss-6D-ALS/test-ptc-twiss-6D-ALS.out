
  ++++++++++++++++++++++++++++++++++++++++++++
  +     MAD-X 5.02.08  (64 bit, Linux)       +
  + Support: mad@cern.ch, http://cern.ch/mad +
  + Release   date: 2015.11.20               +
  + Execution date: 2016.02.01 19:42:39      +
  ++++++++++++++++++++++++++++++++++++++++++++
call, file="../share/ALS/als.seqx";

L1 : drift, L=  2.832695;

L2 : drift, L=  0.45698;

L3 : drift, L=  0.08902;

L4 : drift, L=  0.2155;

L5 : drift, L=  0.219;

L6 : drift, L=  0.107078;

L7 : drift, L=  0.105716;

L8 : drift, L=  0.135904;

L9 : drift, L=  0.2156993;

L10: drift, L= 0.089084;

L11: drift, L= 0.235416;

L12: drift, L= 0.1245;

L13: drift, L= 0.511844;

L14: drift, L= 0.1788541;

L15: drift, L= 0.1788483;

L16: drift, L= 0.511849;

L17: drift, L= 0.1245;

L18: drift, L= 0.235405;

L19: drift, L= 0.089095;

L20: drift, L= 0.2157007;

L21: drift, L= 0.177716;

L22: drift, L= 0.170981;

L23: drift, L= 0.218997;

L24: drift, L=0.215503;

L25: drift, L=0.0890187;

L26: drift, L=0.45698;

L27: drift, L=2.832696;

L27c: drift, L=2.832696-0.2;

ds : drift, L=0.1015;



QF1 : QUADRUPOLE,L=0.344, K1= 2.2474D0+6.447435260914397e-03;

QF2 : QUADRUPOLE,L=0.344, K1= 2.2474;

QD1 : QUADRUPOLE,L=0.187, K1= -2.3368D0-2.593018157427161e-02; 

QD2 : QUADRUPOLE,L=0.187, K1= -2.3368;  

QFA1: QUADRUPOLE,L=0.448, K1= 2.8856;  

QFA2: QUADRUPOLE,L=0.448, K1= 2.8856;  



!!! 1/2 mad-x value

ksf=-41.3355516397069748d0;

ksd=56.2564709584745489d0;



sf:sextupole,l=2*0.1015d0, K2= ksf;

sd: sextupole,l=2*0.1015d0, K2= ksd;



VC5:marker;

ALPHA=0.17453292519943295769236907684886d0;

 

LBEND=0.86621d0;

 

BEND : RBEND, L=LBEND, ANGLE=ALPHA, k1=-0.778741;

BEND1 : RBEND,L= LBEND, ANGLE=ALPHA, k1=-0.778741;

 

CAVM:MARKER;

CAV:RFCAVITY,L=0.2000,VOLT=0.2d0,FREQ=500.;



 sfline: line = (1*sf);

 sdline: line = (1*sd);



SUP1: line =(L1+L2+L3+QF1+VC5+L4+L5+QD1+L6+L7+L8+VC5+BEND+VC5+L9+sfline+L10+

           L11+QFA1+L12+sdline+L13+ 

           L14+BEND+L15+L16+sdline+L17+ 

           QFA2+L18+L19+sfline+L20+BEND+L21+

           L22+QD2+L23+L24+QF2+L25+ 

           L26+VC5+L27+cavm);



SUPb: line=(L1+L2+L3+QF1+VC5+L4+L5+QD1+L6+L7+L8+VC5+BEND+VC5+L9+sfline+L10+

           L11+QFA1+L12+sdline+L13+ 

           L14+BEND+L15+L16+sdline+L17+ 

           QFA2+L18+L19+sfline+L20+BEND1+L21+

           L22+QD2+L23+L24+QF2+L25+ 

           L26+VC5+L27c+cav);





SSTART: marker; !to get compatible with lattice build in F90 code (pure PTC example)



ALS: line = (SSTART + 11*sup1 + supb);





beam, particle = electron, energy = sqrt(1.5*1.5+emass*emass);  use, period=ALS;



!twiss;

!mux_tw = table(twiss, ALS$end, mux);

!muy_tw = table(twiss, ALS$end, muy);

!value, mux_tw, muy_tw;



select,flag=ptc_twiss,clear;

select, flag=ptc_twiss, column=name,s,Energy,x,y,pt,t,

                          beta11,beta22,beta33,mu1,mu2,mu3,disp1,disp2,

                          re11,re12,re13,re14,re15,re16,

                          re21,re22,re23,re24,re25,re26,

                          re31,re32,re33,re34,re35,re36,

                          re41,re42,re43,re44,re45,re46,

                          re51,re52,re53,re54,re55,re56,

                          re61,re62,re63,re64,re65,re66;







ptc_create_universe;

  ptc_create_layout, model=1, method=2, nst=5, exact=false, closed_layout=true;

 MAD-X Beam Parameters
                  Energy : 0.150000E+01
          Kinetic Energy : 0.149949E+01
      Particle Rest Mass : 0.510999E-03
                Momentum : 0.150000E+01
  recomputing with new SECTOR_NMUL and sector_nmul_max            5          10
 Setting MADx with 
     energy           1.50000008703997     
     method                   2
     Num. of steps            5
     charge          -1.00000000000000     
  Length of machine:    196.878135791546     
 The machine is a RING
 ------------------------------------ PTC Survey ------------------------------------
 Before start:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
 Before   end:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
 After  start:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
 After    end:   5.691967490256688E-014  0.000000000000000E+000 -7.716050021144838E-015


!  ptc_printframes, file="ALS.C", format=rootmacro;!this prints my visualisation with ROOT



  !!!!!       T I M E   T R U E ,   T O T P A T H  F A L S E

  ptc_setswitch, debuglevel=1, exact_mis=true, time=true, totalpath=false;

 Setting debug level to           1
maxaccel is found and its value is 1.000000
exact_mis is found and its value is 1.000000
radiation is found and its value is 0.000000
fringe is found and its value is 0.000000
totalpath is found and its value is 0.000000
time is found and its value is 1.000000
nocavity is found and its value is 1.000000
obs_points pro_ptc_setswitch Done
  

!  ptc_script, file="printflatfile.txt";

  

  ptc_twiss, icase=6, no=3, closed_orbit=true;

++++++ info: Zero value of SIGT replaced by 1.
++++++ info: Zero value of SIGE replaced by 1/1000.
 ************ State Summary ****************
 MADTHICK=>KIND =   32  DRIFT-KICK-DRIFT       
  Rectangular Bend: input arc length (rho alpha) 
  Default integration method     2
  Default integration steps      5
 This is an electron (positron actually if charge=1) 
       EXACT_MODEL =  FALSE
       TOTALPATH   =     0
       RADIATION   =  FALSE
       STOCHASTIC  =  FALSE
       ENVELOPE    =  FALSE
       NOCAVITY    =  FALSE
       TIME        =  TRUE 
       FRINGE      =  FALSE
       PARA_IN     =  FALSE
       ONLY_2D     =  FALSE
       ONLY_4D     =  FALSE
       DELTA       =  FALSE
       SPIN        =  FALSE
       MODULATION  =  FALSE
       RAMPING     =  FALSE
       ACCELERATE  =  FALSE

++++++ table: ptc_twiss_summary

            length            alpha_c          alpha_c_p         alpha_c_p2 
       196.8781358      0.00152034687           -1000000           -1000000 

        alpha_c_p3              eta_c           gamma_tr                 q1 
          -1000000     0.001520230817        25.64653265       0.3752128212 

                q2                dq1                dq2                 qs 
      0.4458490684                  0                  0   -2.654417402e-15 

        beta_x_min         beta_x_max         beta_y_min         beta_y_max 
      0.8365220166        12.17178358        1.328095408        22.61686982 

         beta11min          beta11max          beta12min          beta12max 
      0.8365220166        12.17178358    1.667333129e-36    9.281611482e-34 

         beta13min          beta13max          beta21min          beta21max 
   1.535511156e-06      0.01396581595    1.656740233e-38    4.437651316e-35 

         beta22min          beta22max          beta23min          beta23max 
       1.328095408        22.61686982    1.391613138e-33    2.714073223e-28 

         beta31min          beta31max          beta32min          beta32max 
   9.078588374e-06      0.01187969556    4.097811507e-33    7.067508809e-33 

         beta33min          beta33max           disp1min           disp1max 
       65.53499429        67.95294438                  0       0.2319574658 

          disp2min           disp2max           disp3min           disp3max 
     -0.1866966465       0.1866874363   -5.622878742e-16    6.195384101e-16 

          disp4min           disp4max             deltap            orbit_x 
  -3.398492248e-16    3.288495094e-16                  0    -1.54804526e-14 

          orbit_px            orbit_y           orbit_py           orbit_pt 
  -4.063231522e-18                  0                  0   -1.539730901e-12 

         orbit_-cT             xcorms             ycorms            pxcorms 
     -0.1000000058    1.748879194e-13                  0    1.659180151e-13 

           pycorms             tcorms            ptcorms             xcomax 
                 0       0.1000000058    1.539730648e-12   -3.780458408e-15 

            ycomax            pxcomax            pycomax             tcomax 
                 0    2.874849979e-13                  0       0.1000000058 

           ptcomax             xcomin             ycomin            pxcomin 
  -1.539662434e-12   -3.571798474e-13                  0   -2.874739898e-13 

           pycomin             tcomin            ptcomin 
                 0       0.1000000058   -1.539730901e-12 
!  ptc_twiss, icase=6, no=2, betx=10, bety=10,betz=10;

  

!  plot, table=ptc_twiss, vaxis=beta11,beta22, haxis=s;

!  plot, table=ptc_twiss, vaxis=mu3, haxis=s;

!  plot, table=ptc_twiss, vaxis=energy, haxis=s;

  

  write, table=ptc_twiss, file="ALS.6D.ptc.twiss";

  write, table=ptc_twiss_summary, file="ALS.6D.ptc.summ.twiss";

  

ptc_end;





!mux_ptc = table(ptc_twiss, ALS$end, mu1);

!muy_ptc = table(ptc_twiss, ALS$end, mu2);



!value, mux_ptc, muy_ptc;

!value, mux_tw, muy_tw;



!plot, table=ptc_twiss, haxis=s, vaxis=beta11,beta22, colour=100;

!plot, table=ptc_twiss, haxis=s, vaxis=disp1, colour=100;

!plot, table=ptc_twiss, haxis=s, vaxis=disp2, colour=100;

!plot, table=ptc_twiss, haxis=s, vaxis=disp3, colour=100;

!plot, table=ptc_twiss, haxis=s, vaxis=disp4, colour=100;

!plot, table=ptc_twiss, haxis=s, vaxis=x, colour=100;



!removefile, file='ALS.C';

removefile, file='Maxwellian_bend_for_ptc.txt';


  Number of warnings: 0

  ++++++++++++++++++++++++++++++++++++++++++++
  +          MAD-X finished normally         +
  ++++++++++++++++++++++++++++++++++++++++++++
